### DISCUSSION TITLE
Different include paths depending on number of libraries linked?

### DISCUSSION THREAD

**Discussion by HunterZ** [microsoft/vcpkg#37084] [Q&A]
I'm writing an application with vcpkg + cmake that depends on vcpkg's ixwebsocket package, using vcpkg manifest mode and cmake preset mode.

For some reason, if I _only_ link `ixwebsocket::ixwebsocket`, cmake builds with `-isystem .../include/ixwebsocket`, but if I also link against _additional_ vcpkg dependencies (e.g. `nlohmann_json::nlohmann_json`) it also adds in `-isystem .../include`.

Since my code uses `#include <ixwebsocket/xyz.h>`, it breaks depending on how many libraries I'm linking.

Is this really how it's all supposed to work? Am I really supposed to use `#include <xyz.h>` instead of `#include <ixwebsocket/xyz.h>` to make it work in all situations?

It there something in vcpkg or cmake that I can set to tweak this behavior?

Edit: Also note that this behavior contradicts the example here, which says to use `#include <fmt/core.h>`: https://learn.microsoft.com/en-us/vcpkg/get_started/get-started?pivots=shell-cmd#3---add-dependencies-and-project-files

Edit 2: This may be a non-terrible workaround?
```cmake
if(DEFINED VCPKG_INSTALLED_DIR AND DEFINED VCPKG_TARGET_TRIPLET)
  include_directories(AFTER SYSTEM "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/include")
endif()
```

**Comment by Osyotr** [comment]
It looks like a bug in `ixwebsocket`. Their cmake config and pkg-config files don't agree on what include directories are:
For cmake, it's `include/ixwebsocket`: https://github.com/machinezone/IXWebSocket/blob/c5a02f1066fb0fde48f80f51178429a27f689a39/CMakeLists.txt#L275
For pkg-config, it's `include`: https://github.com/machinezone/IXWebSocket/blob/c5a02f1066fb0fde48f80f51178429a27f689a39/ixwebsocket.pc.in#L4
The latter matches their example, the former doesn't.

**Comment by HunterZ** [comment]
Thanks, this all makes sense now. I've opened an issue on that project, but the author recently stated that they're no longer actively maintaining the project, so who knows if/when it might get addressed: https://github.com/machinezone/IXWebSocket/issues/504
